nomos-services/key-management-system/src/keys/ed25519.rs

```diff
@@ -1,13 +1,33 @@
 use bytes::Bytes;
 use ed25519_dalek::{Signature, VerifyingKey, ed25519::signature::Signer as _};
-use serde::{Deserialize, Serialize};
+use nomos_utils::serde::{deserialize_bytes_array, serialize_bytes_array};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use zeroize::ZeroizeOnDrop;
 
 use crate::keys::{errors::KeyError, secured_key::SecuredKey};
 
-#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, ZeroizeOnDrop)]
+#[derive(PartialEq, Eq, Clone, Debug, ZeroizeOnDrop)]
 pub struct Ed25519Key(pub ed25519_dalek::SigningKey);
 
+impl Serialize for Ed25519Key {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serialize_bytes_array(self.0.to_bytes(), serializer)
```

Comment:
@andrussal @danielSanchezQ used `nomos-utils` here, the zk key already handles the yaml case.

Reply:
the nomos-utils handles human readable and wire formats, and it will use hex in case of yaml.

Reply:
Awesome then. Thanks @bacv !